presto-spark-base/src/main/java/com/facebook/presto/spark/planner/PrestoSparkRddFactory.java

```diff
@@ -225,7 +227,15 @@ private static Partitioner createPartitioner(PartitioningHandle partitioning, in
         PrestoSparkTaskDescriptor taskDescriptor = createIntermediateTaskDescriptor(session, tableWriteInfo, fragment);
         SerializedPrestoSparkTaskDescriptor serializedTaskDescriptor = new SerializedPrestoSparkTaskDescriptor(taskDescriptorJsonCodec.toJsonBytes(taskDescriptor));
 
-        if (rddInputs.size() == 1) {
+        if (rddInputs.size() == 0) {
+            checkArgument(fragment.getPartitioning().equals(SINGLE_DISTRIBUTION), "SINGLE_DISTRIBUTION partitioning is expected: %s", fragment.getPartitioning());
+            return sparkContext.parallelize(ImmutableList.of(serializedTaskDescriptor), 1)
+                    .mapPartitionsToPair(createTaskProcessor(
+                            executorFactoryProvider,
+                            taskStatsCollector,
+                            toTaskProcessorBroadcastInputs(broadcastInputs)));
```

Comment:
When the fragment doesn't have any child, it cannot have broadcast input as well right?

Reply:
Theoretically it is possible to get such plans. Think of a broadcast join with VALUES. e.g.: (VALUES JOIN table)